CMakeLists.txt

```diff
@@ -251,6 +251,12 @@ if(ENABLE_AIRTUNES)
   endif()
 endif()
 
+if(CORE_SYSTEM_NAME STREQUAL osx)
+  if(ENABLE_XBMCHELPER)
+    list(APPEND DEP_DEFINES -DHAS_XBMCHELPER)
+  endif()
+endif()
+
```

Comment:
Not a fan of this in this location. Ideally we dont want arch specific stuff scattered through the main CMakeLists.txt file.

At a glance i can see why you have done this, essentially DEP_DEFINES isnt in scope until after core_optional_dep/core_required_dep are called. In this case, i think id look to use SYSTEM_DEFINES and add it as part of https://github.com/xbmc/xbmc/blob/master/cmake/scripts/osx/ExtraTargets.cmake where the inclusion of XBMCHelper is already done.

That'll just remove the CORE_SYSTEM_NAME check, keep the archsetup stuff as you've done, change DEP_DEFINES to SYSTEM_DEFINES